src/commands/apm_cmds/publish.js

```diff
@@ -349,10 +350,15 @@ exports.task = function ({
       title: 'Generate application artifact',
       skip: () => onlyContent && !module.path, // TODO: If onlyContent has been set, get previous version's artifact
       task: async (ctx, task) => {
-        const dir = onlyArtifacts ? cwd : ctx.pathToPublish
-        const artifact = await generateApplicationArtifact(web3, cwd, dir, module, contract, reporter)
-
-        return `Saved artifact in ${dir}/artifact.json`
+        if (pathExistsSync(`${ctx.pathToPublish}/${MANIFEST_FILE}`) &&
+            pathExistsSync(`${ctx.pathToPublish}/${ARTIFACT_FILE}`)) {
+          task.skip('Reusing existing artifact');
```

Comment:
I first thought we might wanna take advantage of the dedicated `skip()` function here, but it seems that API doesn't give us access to `ctx` which we need to look at the correct path. That's my I went for `task.skip()` within the `task()` handler.

Also notice that there's no warning for the user with an option yet to decide whether an existing artifact should be reused or not. I assume this can be easily added with modules like [prompt-choice](https://www.npmjs.com/package/prompt-choices). I first wanted to verify and double check with you if this is the right places for those changes.

Sidenote: I'm unaware of what `onlyContent` in the context of this command means, if you have some insights on that, that'd be great for me for better understanding :)